packages/paste-core/components/form/src/FormLabel.tsx

```diff
@@ -0,0 +1,47 @@
+import * as React from 'react';
+import styled from '@emotion/styled';
+import css from '@styled-system/css';
+import {Box} from '@twilio-paste/box';
+import {Flex} from '@twilio-paste/flex';
+import {Text, safelySpreadTextProps} from '@twilio-paste/text';
+
+export interface FormLabelProps extends React.LabelHTMLAttributes<''> {
+  className?: never;
+  disabled?: boolean;
+  for: string;
+  required?: boolean;
+  style?: never;
+}
+
+// eslint-disable-next-line emotion/syntax-preference
+const StyledRequiredDot = styled(Box)(() =>
+  css({
+    display: 'block',
+    width: '4px',
+    height: '4px',
+    marginRight: 'space20',
+    backgroundColor: 'colorBackgroundDestructive',
+    borderRadius: '50%',
+  })
+);
+
+const FormLabel: React.FC<FormLabelProps> = props => {
+  return (
+    <Flex vAlignContent="center" marginBottom="space10">
+      {props.required ? <StyledRequiredDot /> : null}
```

Comment:
The only downside I can see with this approach is that the required indicator is no longer clickable when it's outside of the `label` element

Reply:
Good catch. Changes so the indicator is part of the label.